From: Mate Kukri <mate.kukri@canonical.com>
Date: Tue, 13 Feb 2024 16:23:06 +0000
Subject: Revert "kern/ieee1275/cmain/ppc64: Introduce flags to identify KVM
 and PowerVM"

This reverts commit 2a9a8518e9df5cb308b3d17bc52ef85ff7be0591.
---
 grub-core/kern/ieee1275/cmain.c  | 8 +-------
 include/grub/ieee1275/ieee1275.h | 4 ----
 2 files changed, 1 insertion(+), 11 deletions(-)

diff --git a/grub-core/kern/ieee1275/cmain.c b/grub-core/kern/ieee1275/cmain.c
index e74de32..400d85d 100644
--- a/grub-core/kern/ieee1275/cmain.c
+++ b/grub-core/kern/ieee1275/cmain.c
@@ -126,10 +126,7 @@ grub_ieee1275_find_options (void)
 
 #if defined(__powerpc__)
       if (grub_strncmp (tmp, "IBM,", 4) == 0)
-	{
-	  grub_ieee1275_set_flag (GRUB_IEEE1275_FLAG_CAN_TRY_CAS_FOR_MORE_MEMORY);
-	  grub_ieee1275_set_flag (GRUB_IEEE1275_FLAG_POWER_VM);
-	}
+	grub_ieee1275_set_flag (GRUB_IEEE1275_FLAG_CAN_TRY_CAS_FOR_MORE_MEMORY);
 #endif
     }
 
@@ -196,9 +193,6 @@ grub_ieee1275_find_options (void)
       grub_ieee1275_set_flag (GRUB_IEEE1275_FLAG_NO_PRE1_5M_CLAIM);
 
       grub_ieee1275_set_flag (GRUB_IEEE1275_FLAG_HAS_CURSORONOFF);
-#if defined(__powerpc__)
-      grub_ieee1275_set_flag (GRUB_IEEE1275_FLAG_POWER_KVM);
-#endif
     }
 }
 
diff --git a/include/grub/ieee1275/ieee1275.h b/include/grub/ieee1275/ieee1275.h
index dddb385..9f669fb 100644
--- a/include/grub/ieee1275/ieee1275.h
+++ b/include/grub/ieee1275/ieee1275.h
@@ -141,10 +141,6 @@ enum grub_ieee1275_flag
    */
   GRUB_IEEE1275_FLAG_CAN_TRY_CAS_FOR_MORE_MEMORY,
 #endif
-
-  GRUB_IEEE1275_FLAG_POWER_VM,
-
-  GRUB_IEEE1275_FLAG_POWER_KVM,
 };
 
 extern int EXPORT_FUNC(grub_ieee1275_test_flag) (enum grub_ieee1275_flag flag);
